include/TPP/Dialect/Tpp/TppOps.td

```diff
@@ -141,39 +207,87 @@ def Tpp_AddOp : Tpp_BinaryOp<"add"> {
   let summary = "Element-wise addition.";
   let description = [{
     The `tpp.add` operation performs element-wise addition on two-dimensional
-    memrefs, writing the result on the output memref. No checks or assumption are
-    made on the input/output arguments so the same memref can be passed both as
-    input and output. It supports broadcast semantic based on Numpy rules. 
+    memrefs or ranked tensors, writing the result on the output memref (or in the
+    result at tensor abstraction). At memref, no checks or assumption are made on
+    the input/output arguments so the same memref can be passed both as input and
+    output. At tensor level the operation produces a new tensor. In both cases, the
+    op supports broadcast semantic see `BroadcastableShape` rules. 
 
     Example:
 
     ```mlir
 
-    // A = A + A
+    // A = A + A - memref abstraction.
     tpp.add ins(%1: memref<2x2xf32>, %1: memref<2x2xf32>)
             outs(%1: memref<2x2xf32>)
 
-    // B = A + B
+    // B = A + B - memref abstraction.
     tpp.add ins(%1: memref<2x2xf32>, %2: memref<2x2xf32>) 
             outs(%2: memref<2x2xf32>)
 
-    // C = A + B
+    // C = A + B - memref abstraction.
     tpp.add ins(%1: memref<2x2xf32>, %2: memref<2x2xf32>)
             outs(%3: memref<2x2xf32>)
 
-    // bcast
+    // bcast.
     tpp.add ins(%1: memref<1x3xf32>, %2: memref<3xf32>)
             outs(%3: memref<3x3xf32>) 
 
+    // tensor abstraction.
+    tpp.add (%1: tensor<3x3xf32>, %2: tensor<3x3xf32>) -> tensor<3x3xf32>
+
     ```
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// Ternary Operations
+//===----------------------------------------------------------------------===//
+
+class Tpp_TernaryOp<string mnemonic, list<Trait> traits = []> :
+  Tpp_Op<mnemonic, !listconcat(traits, [UnitStrideInnerLoop, 
+                                        SameOperandsElementType])> {
+
+  let arguments = (ins TppBrgemmOperand:$shapedA, TppBrgemmOperand:$shapedB, 
+                   TppBrgemmOperand:$shapedC);
```

Comment:
It could also use `Variadic<TppOutputOperand>:$outputs` to be consistent with unary/binary ops and in line with the design doc i.e., support `tpp.matmul ins(%A, %B, %C) outs(%D)`

Reply:
would `%C` be the value we use for initialization?

Reply:
That would be the end goal but we can update to the new syntax separately to not overcomplicate this PR.